### PR TITLE
Update validator reporting logs and metrics for Altair

### LINKED ISSUE #9545: GetValidatorPerformance returns averageInclusionDistance of 0 in v2
# 🐞 Bug Report

### Description

After upgrading to `v2.0.0-rc.1` in `Prater`, I get that the attestation inclusion distance is 0 for all validators, both in the summary posted by the validator and in the information fetched from the beacon chain using `GetValidatorPerformance`. Note that even if the attestation is not produced, a 0 is returned, whereas `^uint(0)` was being returned before.

From the validator logs: Note `averageInclusionDistance="0.00 slots"`
```console
time="2021-09-07 11:39:08" level=info msg="Vote summary since launch" attestationsInclusionPct="100%" averageInclusionDistance="0.00 slots" correctlyVotedHeadPct="86%" correctlyVotedSourcePct="99%" correctlyVotedTargetPct="100%" numberOfEpochs=902 pctChangeCombinedBalance="0.05755%" prefix=validator
```

I also tried querying the beacon node by myself with `GetValidatorPerformance()` and I get that InclusionDistances is always 0. I guess the issue might be somewhere here: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go#L685-L699. The validatorSummary[n].inclusionDistances is empty for Altair.

### Has this worked before in a previous version?

This was working on several previous versions, such as `v1.4.3`, where a valid attestation distance was being returned.

## 🔬 Minimal Reproduction

The issue can be reproduced having a `beacon-chain` running `v2..0.0-rc.1` in Prater and calling `GetValidatorPerformance`.

```go
package main

import (
	"fmt"
	"context"
	"google.golang.org/grpc"
	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
	"github.com/ethereum/go-ethereum/common/hexutil"
)

func main() {
	dialContext, err := grpc.DialContext(context.Background(), "localhost:4000", grpc.WithInsecure())
	if err != nil {
		fmt.Println("error", err)
	}

	beaconClient := ethpb.NewBeaconChainClient(dialContext)

	pubKey := "0x8d7411ab4411ca27d9fd0a9cc05012bbe9be861e8e90ce2909adf5386a05bc484b004006f8dcd0d5207dc3dde0c73dbe"
	key, _ := hexutil.Decode(pubKey)
	pubKeys := make([][]byte, 1)
	pubKeys[0] = key

	req := &ethpb.ValidatorPerformanceRequest{
		PublicKeys: pubKeys,
	}

	valsPerformance, err := beaconClient.GetValidatorPerformance(context.Background(), req)
	if err != nil {
		fmt.Println("error", err)
	}

	fmt.Println("PublicKey:", pubKey)
	fmt.Println(valsPerformance)
}
```
See returned `resp` for different validators.
Inactive key (at the moment of writing)
```console
PublicKey: 0x8d7411ab4411ca27d9fd0a9cc05012bbe9be861e8e90ce2909adf5386a05bc484b004006f8dcd0d5207dc3dde0c73dbe
current_effective_balances:32000000000 inclusion_slots:0 inclusion_distances:0 correctly_voted_source:false correctly_voted_target:false correctly_voted_head:false balances_before_epoch_transition:32334839458 balances_after_epoch_transition:32334824258 public_keys:"\x8dt\x11\xabD\x11\xca'\xd9\xfd\n\x9c\xc0P\x12\xbb龆\x1e\x8e\x90\xce)\t\xad\xf58j\x05\xbcHK\x00@\x06\xf8\xdc\xd0\xd5 }\xc3\xdd\xe0\xc7=\xbe"
```

Active key
```console
PublicKey: 0x8000091c2ae64ee414a54c1cc1fc67dec663408bc636cb86756e0200e41a75c8f86603f104f02c856983d2783116be13
current_effective_balances:32000000000 inclusion_slots:0 inclusion_distances:0 correctly_voted_source:true correctly_voted_target:true correctly_voted_head:true balances_before_epoch_transition:32486753129 balances_after_epoch_transition:32486771083 public_keys:"\x80\x00\t\x1c*\xe6N\xe4\x14\xa5L\x1c\xc1\xfcg\xde\xc6c@\x8b\xc66ˆun\x02\x00\xe4\x1au\xc8\xf8f\x03\xf1\x04\xf0,\x85i\x83\xd2x1\x16\xbe\x13"
```

## 🔥 Error
Not an error per se but an unexpected output `averageInclusionDistance="0.00 slots`.
```console
time="2021-09-07 11:39:08" level=info msg="Vote summary since launch" attestationsInclusionPct="100%" averageInclusionDistance="0.00 slots" correctlyVotedHeadPct="86%" correctlyVotedSourcePct="99%" correctlyVotedTargetPct="100%" numberOfEpochs=902 pctChangeCombinedBalance="0.05755%" prefix=validator
```

## 🌍  Your Environment

**Operating System:** Linux
**What version of Prysm are you running? (Which release)** `v2.0.0-rc.1`
**Anything else relevant (validator index / public key)?** NA

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Inclusion slot and delay are key metrics that users read. This PR is an attempt to replace those metrics with something equally as meaningful for Altair.

**Which issues(s) does this PR fix?**

Fixes #9545

**Other notes for review**

### CODE REVIEW

**Comment by alrevuelta** [comment]
Perhaps add a note in [IndividualVotesRespond](https://github.com/prysmaticlabs/prysm/blob/develop/proto/prysm/v1alpha1/beacon_chain.proto#L894-L897) stating that `inclusion_slot` and `inclusion_distance` are not usable after Altair fork?

**Review by terencechain — DISMISSED** [review]

File: validator/client/metrics.go [hunk 1]
```diff
@@ -258,23 +269,29 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot types.S
 			startBalance := float64(v.startBalances[pubKeyBytes]) / gweiPerEth
 			percentNet := (newBalance - prevBalance) / prevBalance
 			percentSinceStart := (newBalance - startBalance) / startBalance
-			log.WithFields(logrus.Fields{
+
+			previousEpochSummaryFields := logrus.Fields{
```
Comment (potuz) [inline validator/client/metrics.go:273]: I suggest adding here Altair's Inactivity scores

**Comment by potuz** [comment]
Another general comment is that perhaps you want to update the description of the `correctXXX` flags both for logs and metrics. The thing is that after Altair a user might honestly not know if a YES say for `correctlyVotedSource` means that he voted the source or that he got the timely flag for source. I suggest we include the individual bits separately as it would make debugging easier and since this is the only thing that a user would want to see: namely if you got the source right, and within 5 slots, then report yes, and if you got the source right, and got included at 6 slots, then report No.

File: validator/client/metrics.go (hunk 1 above)
Comment (prestonvanloon) [reply validator/client/metrics.go:273]: Thanks! I totally meant to do that!

**Review by terencechain — APPROVED** [review]

**Comment by prestonvanloon** [comment]
This is failing e2e consistently. I'll look into that today.

**Comment by prestonvanloon** [comment]
```
panic: runtime error: index out of range [0] with length 0

goroutine 6554 [running]:
github.com/prysmaticlabs/prysm/validator/client.(*validator).LogValidatorGainsAndLosses(0xc000223040, 0x13f2800, 0xc002381e00, 0x29, 0x7ffe98769e57, 0x3)
        validator/client/metrics.go:267 +0x1ba5
github.com/prysmaticlabs/prysm/validator/client.run.func2(0xc002014930, 0x1406258, 0xc000223040, 0xc00228f460, 0x29, 0xc001dfd960, 0xc00228f490)
        validator/client/runner.go:217 +0x9f
created by github.com/prysmaticlabs/prysm/validator/client.run
        validator/client/runner.go:213 +0x1cd9
```

**Comment by prestonvanloon** [comment]
Ready for another look. I discovered a bug in the previous release that would lead to a panic if the server returned a response where the slices were not the same length, i.e. when Altair fork occurs and some fields are no longer returned.

File: validator/client/metrics.go [hunk 2]
```diff
@@ -408,6 +457,15 @@ func (v *validator) UpdateLogAggregateStats(resp *ethpb.ValidatorPerformanceResp
 		totalPrevBal += v.prevBalance[i]
 	}
 
+	if totalStartBal == 0 ||
```
Comment (terencechain) [inline validator/client/metrics.go:460]: nice!

**Review by terencechain — APPROVED** [review]

**Pull request merged**